cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -96,16 +104,36 @@ public class DefaultAccessEnforcer extends AbstractAccessEnforcer {
     this.metricsCollectionService = metricsCollectionService;
   }
 
+  private void incrementCheckMetric(MetricsContext metricsContext,AuthorizationResult authorizationResult) {
+    switch (authorizationResult.isAuthorized()) {
+      case AUTHORIZED:
+        metricsContext.increment(Constants.Metrics.Authorization.INTERNAL_CHECK_SUCCESS_COUNT, 1);
+        break;
+
+      case UNAUTHORIZED: // Doubt : should we count this in Success as well??
+        metricsContext.increment(Constants.Metrics.Authorization.INTERNAL_CHECK_UNAUTHORIZED_COUNT, 1);
+        break;
+
+      case NOT_REQUIRED:
+        metricsContext.increment(Constants.Metrics.Authorization.INTERNAL_CHECK_NOT_REQUIRED_COUNT, 1);
+        break;
+
+      default:
+        metricsContext.increment(Constants.Metrics.Authorization.INTERNAL_CHECK_FAILURE_COUNT, 1);
+    }
+  }
+
   @Override
-  public void enforce(EntityId entity, Principal principal, Set<? extends Permission> permissions)
-      throws AccessException {
+  public void enforce(EntityId entity, Principal principal, Set<? extends Permission> permissions){
     MetricsContext metricsContext = createEntityIdMetricsContext(entity);
+    AuthorizationResult authorizationResult;
     if (internalAuthEnabled && principal.getFullCredential() != null
         && principal.getFullCredential().getType() == Credential.CredentialType.INTERNAL) {
       LOG.trace("Internal Principal enforce({}, {}, {})", entity, principal, permissions);
       try {
-        internalAccessEnforcer.enforce(entity, principal, permissions);
-        metricsContext.increment(Constants.Metrics.Authorization.INTERNAL_CHECK_SUCCESS_COUNT, 1);
+        authorizationResult = internalAccessEnforcer.enforce(entity, principal, permissions);
+        incrementCheckMetric(metricsContext, authorizationResult);
+        AuthorizationUtil.performAuditLogging(authorizationResult, null); //Figure out SecurityRequestContext
```

Comment:
You don't perform logging here, only collects data. You perform logging later when handler finishes it's work. Also you don't need to pass an instance of SecurityRequestContext. All it's methods are static and storage is thread-local